core/src/main/java/org/apache/accumulo/core/zookeeper/ZooCache.java

```diff
@@ -484,13 +484,18 @@ protected void copyStats(ZcStat userStat, ZcStat cachedStat) {
    * Clears this cache.
    */
   protected void clear() {
-    Preconditions.checkState(!closed);
+    if (closed) {
+      return;
+    }
     nodeCache.clear();
     updateCount.incrementAndGet();
     log.trace("{} cleared all from cache", cacheId);
   }
 
   public void close() {
+    if (closed) {
```

Comment:
Changes here are fine, but I was wondering if it would be better to use a final AtomicBoolean for the closed flag instead of a volatile. The atomic classes are really nice for use cases like this. You could then do something like:

public void close() {
  if (closed.compareAndSet(false, true)) {
    clear();
  }
}
```

Reply:
Oops nevermind, I missed the comment where the AtomicBoolean was switched to a volatile on purpose so ignore.

Reply:
I came here to make the same comment. I know it was changed back on purpose. But the reasoning stated was "that seems fine here". While it may be "fine", I still think the use of compareAndSet is cleaner, and is better than the using volatile directly. Internally, the AtomicBoolean uses a volatile, so it has the same memory effects either way... but the AtomicBoolean is easier to reason about, because it makes it clear what the memory effects are without needing to check the flags on the variable, and compareAndSet is a nice clean API.